### PR TITLE
feat: add request tracing in nginx logs using X-Request-ID header

### PR DESCRIPTION
**Pull request by nadeemshahzad**
Reverts openedx/configuration#6891
The Discovery GoCD pipeline failed with the below error after I merged this PR to enable request tracing in Nginx https://github.com/openedx/configuration/pull/6889

```
fatal: [10.3.120.228]: FAILED! => {
    "changed": true,
    "cmd": [
        "nginx",
        "-t"
    ],
    "delta": "0:00:00.006049",
    "end": "2023-03-09 10:04:53.233906",
    "rc": 1,
    "start": "2023-03-09 10:04:53.227857"
}
```
I debugged the cause and found that in my PR https://github.com/openedx/configuration/pull/6889 I have defined the maps in the virtual host config to set a value for the custom variable `uuid` and used it in `nginx.conf` file to add its value to logs. During Nginx installation, we are testing Nginx configs with Ansible task https://github.com/openedx/configuration/blob/master/playbooks/roles/nginx/tasks/main.yml#L399 which was giving an error for undefined `uuid` variable as virtual host was not yet created where I have defined this variable. So I reverted my PR https://github.com/openedx/configuration/pull/6891 to unblock the pipelines and made this PR with fix to define the `uuid` maps in `nginx.conf` file.
Also made this PR https://github.com/edx/edx-internal/pull/7960 to turn on this feature for our environments.

### CODE REVIEW

File: playbooks/roles/nginx/templates/etc/nginx/nginx.conf.j2 [hunk 1]
```diff
@@ -35,6 +35,21 @@ http {
         include /etc/nginx/mime.types;
         default_type application/octet-stream;
 
+        {% if NGINX_ENABLE_UUID %}
```
Comment (matthugs) [inline playbooks/roles/nginx/templates/etc/nginx/nginx.conf.j2]:
```suggestion
        {%- if NGINX_ENABLE_UUID %}
```
you can avoid adding unnecessary whitespace to these files using jinja syntax like this

the minus sign on one side or the other essentially means "strip out whitespace to this side of this block", so that we don't have the indentation of the `{% if %}` block itself included in the rendered output.

File: playbooks/roles/nginx/templates/etc/nginx/nginx.conf.j2 [hunk 2]
```diff
@@ -35,6 +35,21 @@ http {
...
+        # Maps to trace requests and identifying them across different components. First map creates a new variable named
+        # trace_id, which is set to either the value of X-REQUEST-ID or Cloudflare CF-ray headers depending on whether 
+        # http_x_request_id is empty or not. Second map creates a new variable named uuid, which is set to either the 
+        # value of trace_id or request_id, depending on whether trace_id is set or not in first map.
+        map $http_x_request_id $trace_id {
+          ""        "${http_cf_ray}";
+          default   "${http_x_request_id}";
+        }
+        map $trace_id $uuid {
```
Comment (matthugs) [inline playbooks/roles/nginx/templates/etc/nginx/nginx.conf.j2]: I think uuid may be a bad choice of name for this variable because in some cases it will take the value of the CF-Ray header, which is not formatted as a UUID.

File: playbooks/roles/nginx/templates/etc/nginx/nginx.conf.j2 [hunk 3]
```diff
@@ -35,6 +35,21 @@ http {
         include /etc/nginx/mime.types;
         default_type application/octet-stream;
 
+        {% if NGINX_ENABLE_UUID %}
+        # Maps to trace requests and identifying them across different components. First map creates a new variable named
```
Comment (matthugs) [inline playbooks/roles/nginx/templates/etc/nginx/nginx.conf.j2]: [nit] I think we should probably avoid the word "map" in this comment, because while that's an nginx feature we're using here it's not a good word to use to describe what we're doing at a higher level. I would probably use words like "ternary" or "setting a value conditionally" to explain this. "Map" usually means something very different, i.e. transforming values drawn from an array and outputting an array of "mapped" values (which is how the word is used in a functional programming context).

**Review by matthugs — APPROVED** [review]

File: playbooks/roles/nginx/templates/etc/nginx/nginx.conf.j2 (hunk 1 above)
Comment (nadeemshahzad) [reply playbooks/roles/nginx/templates/etc/nginx/nginx.conf.j2]: thanks, @matthugs I've updated the template with the control character to remove space after.

File: playbooks/roles/nginx/templates/etc/nginx/nginx.conf.j2 (hunk 2 above)
Comment (nadeemshahzad) [reply playbooks/roles/nginx/templates/etc/nginx/nginx.conf.j2]: Replaced variable `uuid` with `request_tracking_id`

File: playbooks/roles/nginx/templates/etc/nginx/nginx.conf.j2 (hunk 3 above)
Comment (nadeemshahzad) [reply playbooks/roles/nginx/templates/etc/nginx/nginx.conf.j2]: Addressed and updated the comment with note.

Setting variables based on HTTP headers to track and differentiate requests across components: the first creates a new
 variable called trace_id based on X-REQUEST-ID or Cloudflare CF-ray headers, while the second creates a variable named
  request_tracking_id based on whether trace_id is set or not.
